css: Fix search bar width in right sidebar.

SameepAher opened the pull request:
[CZO](https://chat.zulip.org/#narrow/stream/101-design/topic/search.20people.20ui.20not.20looking.20good.20on.20small.20width.20devices)

**Screenshots:**

**before -**

![before](https://user-images.githubusercontent.com/96468766/220564022-b83bc5fa-1b19-459c-9ca3-08a7f940bac2.png)

**after -**

![screen-1](https://user-images.githubusercontent.com/96468766/220564110-413b442f-d640-49e7-86c5-bd0b195dfd72.png)

![screen-2](https://user-images.githubusercontent.com/96468766/220564114-c12a03da-d6d2-45ee-9ebe-2bfc1ea38173.png)

![screen-3](https://user-images.githubusercontent.com/96468766/220564118-735e8629-0b93-4ce8-b9f2-f7dc93132ffa.png)

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

timabbott commented:
@amanagr can you review this? I imagine this could be related to the floating recipient bar changes, and am not sure if there's a more correct change we should be making than just setting a `padding-right` here.

amanagr commented:
We have this very old CSS that doesn't seem to be working correctly and likely is broken from a long time, just removing this seems to be the right fix.
```css
@media (width < $md_min) {
    #user_search_section .user-list-filter {
        /* input should be 100% - 6px padding x2 - 1px border x2. */
        width: calc(100% - 12px - 2px);
    }
}
```
@SameepAher can you add this change?

SameepAher commented:
> We have this very old CSS that doesn't seem to be working correctly and likely is broken from a long time, just removing this seems to be the right fix.
> 
> ```css
> @media (width < $md_min) {
>     #user_search_section .user-list-filter {
>         /* input should be 100% - 6px padding x2 - 1px border x2. */
>         width: calc(100% - 12px - 2px);
>     }
> }
> ```
> 
> @SameepAher can you add this change?

Added this change. Seems to be working fine.

Also updated screenshots in the PR description.

timabbott commented:
Merged, thanks @SameepAher!
